spec/helpers/application_helper_spec.rb

```diff
@@ -134,4 +134,64 @@ def current_organization; end
       end
     end
   end
+
+  describe "#set_default_location for purchase" do
+    helper do
+      def current_organization; end
+    end
+
+    before(:each) do
+      allow(helper).to receive(:current_organization).and_return(organization)
+    end
+
+    context "returns storage_location_id if present" do
+      let(:purchase) { build(:purchase, storage_location_id: 2) }
+      subject { helper.set_default_location(purchase) }
+
+      it { is_expected.to eq(2) }
+    end
+
+    context "returns current_organization intake_location if storage_location_id is not present" do
+      let(:organization) { build(:organization, intake_location: 1) }
+      let(:purchase) { build(:purchase, storage_location_id: nil) }
+
+      before do
+        allow(helper).to receive(:current_organization).and_return(organization)
+      end
+
+      subject { helper.set_default_location(purchase) }
+
+      it { is_expected.to eq(1) }
+    end
+  end
+
+  describe "#set_default_location for donation" do
```

Comment:
I don't think we need to copy the same test for all things that have storage locations - all we're doing is referencing the `storage_location_id` of that thing.

Reply:
I've opted to keep one test as per commit [2716706](https://github.com/rubyforgood/human-essentials/pull/4722/commits/27167069b67440b7361a3a213d495832fe6899f7) so that we guarantee the functionality of the method and any subsequent changes to it.

Let me know if that is in keeping with what you were suggesting or your proposal was to remove the test altogether.